sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/ServiceEndpointTest.java

```diff
@@ -0,0 +1,66 @@
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.AsyncDocumentClient;
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.TestConfigurations;
+import com.azure.cosmos.models.CosmosItemResponse;
+import com.azure.cosmos.rx.TestSuiteBase;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ServiceEndpointTest extends TestSuiteBase {
+
+    @DataProvider
+    private Object[][] serviceEndpointArgProvider() {
+        return new Object[][] {
+            { TestConfigurations.HOST.substring(0,TestConfigurations.HOST.length() - 1), true }, // https://localhost:8080
+            { TestConfigurations.HOST, true }, // https://localhost:8080/
+            { TestConfigurations.HOST.replace("https://", ""), true }, // localhost:8080/
+            { TestConfigurations.HOST.replace("https://", "").replace("/", ""), true },  // localhost:8081
+            { TestConfigurations.HOST + ";AccountKey=" + TestConfigurations.MASTER_KEY + ";", true }, // https://localhost:8081/;AccountKey=<secret>;
+            { TestConfigurations.HOST + ";AccountKey=" + TestConfigurations.MASTER_KEY, true}, // https://localhost:8081/;AccountKey=<secret>
+            { TestConfigurations.HOST + ";" + TestConfigurations.MASTER_KEY + ";", true }, // https://localhost:8081/;<secret>;
+            { TestConfigurations.HOST + TestConfigurations.MASTER_KEY + ";", true }, // https://localhost:8081/<secret>;
+            { TestConfigurations.HOST + TestConfigurations.MASTER_KEY, true }, // https://localhost:8081/<secret>
+            { "AccountEndpoint=" + TestConfigurations.HOST + ";AccountKey=" + TestConfigurations.MASTER_KEY + ";", false }, // AccountEndpoint=https://localhost:8081/;AccountKey=<secret>;
```

Comment:
This is false right now, but should it work? Why do we expect to receive `;AccountSecret=...`, but not the `AccountEndpoint=` part?